### ISSUE TITLE
CNN architecture the same but getting different results

### ISSUE THREAD

**Issue by ericj0114** [keras-team/keras#10973]
I have a CNN that saves the bottleneck features of the training and test data with the VGG16 architecture, then uploads the features to my custom fully connected layers to classify the images. 

 
    #create data augmentations for training set; helps reduce overfitting and find more features
    train_datagen = ImageDataGenerator(rescale=1./255,
                            shear_range = 0.2,
                            zoom_range = 0.2,
                            horizontal_flip=True)
    
    #use ImageDataGenerator to upload validation images; data augmentation not necessary for 
    validating process
    val_datagen = ImageDataGenerator(rescale=1./255)
    
    #load VGG16 model, pretrained on imagenet database
    model = applications.VGG16(include_top=False, weights='imagenet')
    
    #generator to load images into NN
    train_generator = train_datagen.flow_from_directory(
            train_dir,
            target_size=(img_width, img_height),
            batch_size=batch_size,
            class_mode=None,
            shuffle=False)
    
    #total number of images used for training data
    num_train = len(train_generator.filenames)
    
    #save features to numpy array file so features do not overload memory
    bottleneck_features_train = model.predict_generator(train_generator, num_train // batch_size)

    val_generator = val_datagen.flow_from_directory(
            val_dir,
            target_size=(img_width, img_height),
            batch_size=batch_size,
            class_mode=None,
            shuffle=False)
    
    num_val = len(val_generator.filenames)
    
    bottleneck_features_validation = model.predict_generator(val_generator, num_val // batch_size)`

   

    #used to retrieve the labels of the images
    label_datagen = ImageDataGenerator(rescale=1./255)
    
    #generators can create class labels for each image in either 
    train_label_generator = label_datagen.flow_from_directory(  
        train_dir,  
        target_size=(img_width, img_height),  
        batch_size=batch_size,  
        class_mode=None,  
        shuffle=False)  
    
    #total number of images used for training data
    num_train = len(train_label_generator.filenames)
    
    #load features from VGG16 and pair each image with corresponding label (0 for normal, 1 for pneumonia)
    #train_data = np.load('xray/bottleneck_features_train.npy')
    #get the class labels generated by train_label_generator 
    train_labels = train_label_generator.classes
    
    val_label_generator = label_datagen.flow_from_directory(  
        val_dir,  
        target_size=(img_width, img_height),  
        batch_size=batch_size,  
        class_mode=None,  
        shuffle=False)
    
    num_val = len(val_label_generator.filenames)
    
    #val_data = np.load('xray/bottleneck_features_validation.npy')
    val_labels = val_label_generator.classes
    
    #create fully connected layers, replacing the ones cut off from the VGG16 model
    model = Sequential()
    #converts model's expected input dimensions to same shape as bottleneck feature arrays 
    model.add(Flatten(input_shape=bottleneck_features_train.shape[1:]))
    #ignores a fraction of input neurons so they do not become co-dependent on each other; helps prevent overfitting
    model.add(Dropout(0.7))
    #normal fully-connected layer with relu activation. Replaces all negative inputs with 0 and does not fire neuron,
    #creating a lighetr network
    model.add(Dense(128, activation='relu'))
    model.add(Dropout(0.7))
    #output layer to classify 0 or 1
    model.add(Dense(1, activation='sigmoid'))

    #compile model and specify which optimizer and loss function to use
    #optimizer used to update the weights to optimal values; adam optimizer maintains seperate learning rates
    #for each weight and updates accordingly
    
    #cross-entropy function measures the ability of model to correctly classify 0 or 1
    model.compile(optimizer=optimizers.Adam(lr=0.0007), loss='binary_crossentropy', metrics=['accuracy'])

    #used to stop training if NN shows no improvement for 5 epochs
    early_stop = EarlyStopping(monitor='val_loss', min_delta=0.01, patience=5, verbose=1)
    
    #checks each epoch as it runs and saves the weight file from the model with the lowest validation loss
    checkpointer = ModelCheckpoint(filepath=top_model_weights_dir, verbose=1, save_best_only=True)

    #fit the model to the data
    history = model.fit(bottleneck_features_train, train_labels,
            epochs=epochs,
            batch_size=batch_size,
            callbacks = [early_stop, checkpointer],
            verbose=2,
            validation_data=(bottleneck_features_validation, val_labels))`
After calling train_top_model(), the CNN gets an 86% accuracy after around 10 epochs.

However, when I try implementing this architecture in by building the fully connected layers directly on top of the VGG16 layers, The network gets stuck at a val_acc of 0.5000 and basically does not train. Are there any issues with the code? 

    epochs = 10
    batch_size = 20

    train_datagen = ImageDataGenerator(rescale=1./255,
                            shear_range = 0.2,
                            zoom_range = 0.2,
                            horizontal_flip=True)

    val_datagen = ImageDataGenerator(rescale=1./255)

    train_generator = train_datagen.flow_from_directory(
            train_dir,
            target_size=(img_width, img_height),
            batch_size=batch_size,
            class_mode='binary',
            shuffle=False)

    num_train = len(train_generator.filenames)

    val_generator = val_datagen.flow_from_directory(
            val_dir,
            target_size=(img_width, img_height),
            batch_size=batch_size,
            class_mode='binary',
            shuffle=False)

    num_val = len(val_generator.filenames)`

    base_model = applications.VGG16(weights='imagenet', include_top=False, input_shape=(img_width, 
    img_height, 3))

    x = base_model.output
    x = Flatten()(x)
    x = Dropout(0.7)(x)
    x = Dense(128, activation='relu')(x)
    x = Dropout(0.7)(x)
    predictions = Dense(1, activation='sigmoid')(x)

    model = Model(inputs=base_model.input, outputs=predictions)

    for layer in model.layers[:19]:
    layer.trainable = False

    checkpointer = ModelCheckpoint(filepath=top_model_weights_dir, verbose=1, save_best_only=True)

    model.compile(optimizer=optimizers.Adam(lr=0.0007), loss='binary_crossentropy', metrics= 
    ['accuracy'])

    history = model.fit_generator(train_generator, 
                              steps_per_epoch=(num_train//batch_size), 
                              validation_data=val_generator,
                              validation_steps=(num_val//batch_size),
                              callbacks=[checkpointer], 
                              verbose=1,
                              epochs=epochs)

**Comment by kashyapraval** [comment]
for layer in model.layers[:19]:
layer.trainable = False
is this indented as intended?(may be a copy-paste issue)
Do you intend to set all layers as false? Why are 19 layers being set to non-trainable?
